src/components/ui/ContextMenu.jsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export default class ContextMenu extends React.Component {
+  static propTypes = {
+    items: PropTypes.arrayOf(PropTypes.shape({
+      label: PropTypes.string.isRequired,
+      icon: PropTypes.string.isRequired,
+      action: PropTypes.func.isRequired,
+    })).isRequired,
```

Comment:
is... is this code checking that 3 isRequired properties are true?
or that at least one is true?
in any case, it could be replaced by something clearer...

Reply:
It means that the `items` array itself is required, and that each of its object elements must have these three fields, `label`, `icon` and `action`.